### ISSUE TITLE
Not-hardcoded provideCompletionItems

### ISSUE THREAD

**Issue by alrod** [Microsoft/monaco-editor#24]
Is it possible to call web service in provideCompletionItems to return not-hardcoded/dynamic provideCompletetionItems? 

monaco.languages.registerCompletionItemProvider('mySpecialLanguage', {
	provideCompletionItems: () => {
		return [
			{
				label: 'simpleText',
				kind: monaco.languages.CompletionItemKind.Text
			}, {
				label: 'testing',
				kind: monaco.languages.CompletionItemKind.Keyword,
				insertText:'testing({{condition}})'
			},
			{
				label: 'ifelse',
				kind: monaco.languages.CompletionItemKind.Snippet,
				insertText: [
					'if ({{condition}}) {',
					'\t{{}}',
					'} else {',
					'\t',
					'}'
				].join('\n'),
				documentation: 'If-Else Statement'
			}
		]
	}
});
https://microsoft.github.io/monaco-editor/playground.html#extending-language-services-custom-languages

**Comment by stofte** [comment]
The API supports returning a promise, inside of which you can perform your lookup.

return new Promise(succ => {
    ajax.then(jsonList => {
        succ(jsonList)
    });
});

instead of returning your array directly.

**Comment by alexandrudima** [comment]
@alrod I have created a sample (it uses a hover provider, but the idea is the same, exactly as @stofte commented):

https://microsoft.github.io/monaco-editor/playground.html#extending-language-services-hover-provider-example

**Comment by alrod** [comment]
@alexandrudima , @stofte thanks guys! it works!
